spring-cloud-function-gateway/src/main/java/org/springframework/cloud/function/gateway/ProxyExchange.java

```diff
@@ -0,0 +1,611 @@
+        ResponseExtractor<ResponseEntity<T>> responseExtractor = rest
+                .responseEntityExtractor(type);
+        return rest.execute(requestEntity.getUrl(), requestEntity.getMethod(),
+                requestCallback, responseExtractor);
+    }
+
+    private BodyBuilder headers(BodyBuilder builder) {
+        Set<String> sensitive = this.sensitive;
+        if (sensitive == null) {
+            sensitive = DEFAULT_SENSITIVE;
+        }
+        for (String name : headers.keySet()) {
+            if (sensitive.contains(name.toLowerCase())) {
+                continue;
+            }
+            builder.header(name, headers.get(name).toArray(new String[0]));
+        }
+        return builder;
+    }
+
+    private Object body() {
+        if (body != null) {
+            return body;
+        }
+        body = getRequestBody();
+        return body;
+    }
+
+    /**
+     * Search for the request body if it was already deserialized using
+     * <code>@RequestBody</code>. If it is not found then deserialize it in the same way
+     * that it would have been for a <code>@RequestBody</code>.
+     * 
+     * @return the request body
+     */
+    private Object getRequestBody() {
+        for (String key : mavContainer.getModel().keySet()) {
+            if (key.startsWith(BindingResult.MODEL_KEY_PREFIX)) {
+                BindingResult result = (BindingResult) mavContainer.getModel().get(key);
+                return result.getTarget();
+            }
+        }
+        MethodParameter input = new MethodParameter(
+                ClassUtils.getMethod(BodyGrabber.class, "body", Object.class), 0);
+        try {
+            delegate.resolveArgument(input, mavContainer, webRequest, binderFactory);
+        }
+        catch (Exception e) {
+            throw new IllegalStateException("Cannot resolve body", e);
+        }
+        String name = Conventions.getVariableNameForParameter(input);
+        BindingResult result = (BindingResult) mavContainer.getModel()
+                .get(BindingResult.MODEL_KEY_PREFIX + name);
+        return result.getTarget();
+    }
+
+    private NestedTemplate createTemplate(RestTemplate input) {
+        NestedTemplate rest = new NestedTemplate();
+        rest.setMessageConverters(input.getMessageConverters());
+        rest.setErrorHandler(input.getErrorHandler());
+        rest.setDefaultUriVariables(
+                ((AbstractUriTemplateHandler) input.getUriTemplateHandler())
+                        .getDefaultUriVariables());
+        rest.setRequestFactory(input.getRequestFactory());
+        rest.setInterceptors(input.getInterceptors());
+        return rest;
+    }
+
+    /**
+     * A special {@link RestTemplate} that knows about the {@link Type} of its response
+     * body explicitly (rather than through a {@link ParameterizedTypeReference}, which is
+     * the only way to access this feature in a regular template).
+     *
+     */
+    class NestedTemplate extends RestTemplate {
```

Comment:
I'm not seeing what this is adding... both methods are simply delegating to their corresponding `super` method.